### ISSUE TITLE
Add metadata information in tag on PD

### ISSUE THREAD

**Issue by davidz627** [kubernetes-sigs/gcp-compute-persistent-disk-csi-driver#387]
info like namespace and pvc name (check with feature parity for in-tree plugin)

**Comment by davidz627** [comment]
In-tree plugin provisions with `Description` field example:
```
Description
{"kubernetes.io/created-for/pv/name":"pvc-6a10b78e-4cae-4c9e-8dd2-1cb2c60211e1","kubernetes.io/created-for/pvc/name":"test-disk","kubernetes.io/created-for/pvc/namespace":"default"}
```

No such thing as a "tag" AFAIK. Do people really rely on json description text? I guess we should continue this for backwards compatibility reasons. Considering adding some more description fields to specify the driver name that provisioned to volume.

/cc @msau42

**Comment by msau42** [comment]
What about "labels"? https://cloud.google.com/kubernetes-engine/docs/how-to/creating-managing-labels

**Comment by davidz627** [comment]
IIRK labels are new and basically what we were trying to approximate with the description field before. Labels are not used by in-tree currently. They are empty on my test PD. 

IIUC description should stay the same because of backwards compatibility (who is actually depending on this functionality...?). Do we even have any expectations for backwards compatibility for a freeform text field meant for human-readable descriptions? 

We could duplicate whats in the description into the labels but that's almost uglier than only having them in the description. However, I would prefer to push to put this information *only* in the labels and break the descriptions field.

**Comment by msau42** [comment]
It has the possibility to break users once migration is on though.

**Comment by davidz627** [comment]
Depends on: https://github.com/kubernetes-csi/external-provisioner/issues/370

**Comment by davidz627** [comment]
depends on: https://github.com/kubernetes-csi/external-provisioner/pull/399

**Comment by verult** [comment]
kubernetes-csi/external-provisioner#399 now available in v1.6.0

**Comment by shay-berman** [comment]
what is the status of this ticket?
I would like to know when CSI driver will be GA in GKE?
